test/e2e/prometheus_test.go

```diff
@@ -1339,6 +1339,155 @@ func TestPrometheusGetBasicAuthSecret(t *testing.T) {
 
 }
 
+// testOperatorNSScope tests the multi namespace feature of the Prometheus Operator.
+// It checks whether it ignores rules that are not in the watched namespaces of the
+// Prometheus Operator. The Prometheus Operator internally treats watching a
+// single namespace different than watching multiple namespaces, hence the two
+// sub-tests.
+func testOperatorNSScope(t *testing.T) {
+	name := "test"
+	firtAlertName := "firstAlert"
+	secondAlertName := "secondAlert"
+
+	t.Run("SingleNS", func(t *testing.T) {
```

Comment:
I have split this test up into two subtests. One with the Prometheus Operator watching a single namespace, one watching two namespaces.